tests/integration/test_zarr_to_iris.py

```diff
@@ -0,0 +1,98 @@
+"""Test conversion of remote and local Zarr store to iris Cube."""
+from importlib.resources import files as importlib_files
+from pathlib import Path
+
+import iris
+import pytest
+import xarray as xr
+import ncdata
+import ncdata.iris_xarray
+import zarr
+
+
+def _return_kwargs():
+    time_coder = xr.coders.CFDatetimeCoder(use_cftime=True)
+    xr_kwargs = {
+            "consolidated": True,
+            "decode_times": time_coder,
+            "engine": "zarr",
+            "chunks": {},
+            "backend_kwargs": {},
+    }
+
+    return xr_kwargs
+
+
+def _run_checks(cube):
+    """Run some standard checks."""
+    assert cube.var_name == "q"
+    assert cube.standard_name == "specific_humidity"
+    assert cube.long_name is None
+    coords = cube.coords()
+    coord_names = [coord.standard_name for coord in coords]
+    assert "longitude" in coord_names
+    assert "latitude" in coord_names
+
+
+def test_load_zarr2_local():
+    """Test loading a Zarr2 store from local FS."""
+    zarr_path = (
+        Path(importlib_files("tests"))
+        / "zarr-sample-data"
+        / "example_field_0.zarr2"
+    )
+
+    xr_kwargs = _return_kwargs()
+    zarr_xr = xr.open_dataset(zarr_path, **xr_kwargs)
+    zarr_xr.unify_chunks()
+
+    conversion_func = ncdata.iris_xarray.cubes_from_xarray
+    cubes = conversion_func(zarr_xr)
+
+    assert len(cubes) == 1
+    cube = cubes[0]
+    _run_checks(cube)
+
+
+def test_load_zarr3_local():
+    """Test loading a Zarr3 store from local FS."""
+    zarr_path = (
+        Path(importlib_files("tests"))
+        / "zarr-sample-data"
+        / "example_field_0.zarr3"
+    )
+
+    xr_kwargs = _return_kwargs()
+    zarr_xr = xr.open_dataset(zarr_path, **xr_kwargs)
+    zarr_xr.unify_chunks()
+
+    conversion_func = ncdata.iris_xarray.cubes_from_xarray
+    cubes = conversion_func(zarr_xr)
+
+    assert len(cubes) == 1
+    cube = cubes[0]
+    _run_checks(cube)
+
+
+def test_load_remote_zarr():
```

Comment:
```suggestion
#TODO: imports at the top
import requests
from requests.exceptions import RequestException

def _is_url_ok(url):
    try:
        req = requests.get(url)
        result = req.status_code == 200
    except RequestException as err:
        result = False
    return result

S3_TEST_PATH = (
    "https://uor-aces-o.s3-ext.jc.rl.ac.uk/"
    "esmvaltool-zarr/pr_Amon_CNRM-ESM2-1_02Kpd-11_r1i1p2f2_gr_200601-220112.zarr3"
)
_S3_accessible = _is_url_ok(S3_TEST_PATH)

@pytest.mark.skipif(not _S3_accessible, reason="S3 url not accessible")
def test_load_remote_zarr():
```

I presume we **don't** really have a problem with the dataset not being stable and available?

I'm only suggesting this addition because, _here at the MetOffice_, we can't run this test locally since we don't have suitable network permission.

Reply:
indeed, am aware of this (if only your sysadmins updated SSL - that's actually something you guys should strongly suggest, an old SSL is a security risk); this is the bucket we're using in ESMValTool, so no worries at all, it'll not go anywhere anytime soon (there may be occasional downtimes when CEDA update their S3 software though, but not something very frequent at all)

Reply:
Great, you made it work !